README.md

```diff
@@ -41,17 +41,97 @@ Note: *1.0.0 has major breaking changes, you will need to update any automations
 
 ## Using Services
 
-Available services are `opensprinkler.run` for programs and stations, and `opensprinkler.stop` for stations or controller (to stop all stations).
+Available services are `opensprinkler.run` for programs, stations and controllers (for running once program), and `opensprinkler.stop` for stations or controller (to stop all stations).
+
+### Run Examples
+
+#### Run Program Example
 
 ```yaml
 service: opensprinkler.run
 data:
-  entity_id: switch.station_name # Switches or sensors for programs or stations
-  run_seconds: 60 # Seconds to run for (optional, defaults to 60 seconds, stations only)
+  entity_id: switch.program_name # Switches or sensors for programs
 ```
 
+#### Run Station Example
+
+```yaml
+service: opensprinkler.run
+data:
+  entity_id: switch.station_name # Switches or sensors for stations
+  run_seconds: 60 # Seconds to run (optional, defaults to 60 seconds)
+```
+
+#### Run Once Program Example
+
+To run an once program, the run seconds can either be a list of seconds per station or a list of index and second pairs.
+The following two examples are equivalent.
+
+```yaml
+service: opensprinkler.run
+data:
+  entity_id: switch.controller_name # Switches or sensors for controller
+  run_seconds: # Seconds to run for each station (required)
+    - 60
+    - 0
+    - 30
+```
+
+```yaml
+service: opensprinkler.run
+data:
+  entity_id: switch.controller_name # Switches or sensors for controller
+  run_seconds: # Station and index and seconds pairs (required)
+    - index: 1
```

Comment:
It follows the station index so 1 indexed.

I couldn't figure out how to do variable keys in the automation template so I think this way is better

Reply:
Stations are 0 indexed for the underlying method and more importantly the station attributes are 0 indexed. I think we’re better to use that.

Reply:
I can try out the template logic and see what I get I suppose. Only question is whether hass allows mixed input types in the service definition?

Reply:
You are right about the index, I don't know why I thought it was 1 index, I think I got confused with the default station names.

The type is fine, but having variable keys might not be possible. I guess it could be just declared as an unstructured object. I don't think I have ever seen HA config or service data done in that way though.

Reply:
Yeah the indices are unfortunate because one of the data points (don’t recall which right now) is 1 indexed. Crazy.

I’ll look into the input structure and give it a whirl and see what I get.

Reply:
Yeah, hass is puking before it ever gets invoked due to the schema validation. Can you send over what the schema validation would look like to support:

```
entity_id: switch.opensprinkler_enabled
run_seconds:
  1: 30
  10: 60
```

Reply:
I think you can just add `dict`, I couldn't figure out any other nice way:
```
--- a/custom_components/opensprinkler/const.py
+++ b/custom_components/opensprinkler/const.py
@@ -24,6 +24,7 @@ SCHEMA_SERVICE_RUN = {
         cv.ensure_list(cv.positive_int),
         cv.ensure_list(SCHEMA_SERVICE_RUN_SECONDS),
         cv.positive_int,
+      dict,
     )
 }
 SCHEMA_SERVICE_STOP = {}
```

Reply:
or try: 
```
--- a/custom_components/opensprinkler/const.py
+++ b/custom_components/opensprinkler/const.py
@@ -24,6 +24,7 @@ SCHEMA_SERVICE_RUN = {
         cv.ensure_list(cv.positive_int),
         cv.ensure_list(SCHEMA_SERVICE_RUN_SECONDS),
         cv.positive_int,
+        vol.Schema({}, extra=vol.ALLOW_EXTRA),
     )
 }
 SCHEMA_SERVICE_STOP = {}
```

Reply:
supporting that structure should not be a problem but I don't think its possible to use templating in the key name so if that is used the index has to be hard coded in the script/automation

Reply:
I pushed a commit that will support the dict as well so now it supports all three structures

Reply:
Perfect!